lib/ecto_adapters_dynamodb/query.ex

```diff
@@ -90,58 +100,89 @@ defmodule Ecto.Adapters.DynamoDB.Query do
         query = construct_search(index, search, opts)
 
         if op == :in do
-          responses_element = "Responses"
-          response_map = %{responses_element => %{table => []}}
+          response_map = %{@responses_element_key => %{table => []}}
 
           Enum.reduce(hash_values, response_map, fn(hash_value, acc) ->
             # When receiving a list of values to query on, construct a custom query for each of those values to pass into do_fetch_recursive/1.
-            %{"Items" => items} = Kernel.put_in(query, [:expression_attribute_values, :hash_key], hash_value)
+            %{"Items" => items} = put_in(query, [:expression_attribute_values, :hash_key], hash_value)
                                   |> (do_fetch_recursive).()
 
-            Kernel.put_in(acc, [responses_element, table], acc[responses_element][table] ++ items)
+            put_in(acc, [@responses_element_key, table], acc[@responses_element_key][table] ++ items)
           end)
         else
           do_fetch_recursive.(query)
         end
       :scan ->
         maybe_scan(table, search, opts)
     end
+  end
 
-    results
+  @spec filter(dynamo_response,  non_indexed_filters, String.t) :: dynamo_response | %{}
```

Comment:
How is type `dynamo_response` different from `%{}` (i.e., why the `|` ?)

Reply:
Perhaps I misunderstand... `dynamo_response` is defined as `%{required(String.t) => term}`.  `filter` can return a `dynamo_response`, or in the case of a query for a single result that fails to pass the filter, an empty map.

Does our definition of `dynamo_response` cover an empty map?  My understanding of the typespec docs is that an empty map has a separate definition - https://hexdocs.pm/elixir/typespecs.html

If I'm correct, then I think we'd also need to add an empty map to the spec for `get_item`. If I'm incorrect, then I'll remove the empty map from `filter`'s spec.

Reply:
What I meant was that they both look like type map (I assume there is such
a thing).

Reply:
Oh, I see, they seem like different types, as you explained.

Reply:
There is a general `map()` type for any map, but since we had the distinct `dynamo_response` definition, I figured I'd roll with that, though it also necessitates specifying the empty map.